testing/simulated/simulated_test.go

```diff
@@ -0,0 +1,261 @@
+// TITLE.
+
+package simulated_test
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/berachain/beacon-kit/beacon/blockchain"
+	"github.com/berachain/beacon-kit/consensus/cometbft/service/encoding"
+	"github.com/berachain/beacon-kit/engine-primitives/errors"
+	"github.com/berachain/beacon-kit/log/phuslu"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/constants"
+	mathpkg "github.com/berachain/beacon-kit/primitives/math"
+	"github.com/berachain/beacon-kit/testing/simulated"
+	"github.com/berachain/beacon-kit/testing/simulated/execution"
+	"github.com/cometbft/cometbft/abci/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// SimulatedSuite defines our test suite for the simulated Comet component.
+type SimulatedSuite struct {
+	suite.Suite
+	// Embedded shared accessors for convenience.
+	simulated.SharedAccessors
+
+	SimComet              *simulated.SimComet
+	LogBuffer             *bytes.Buffer
+	GenesisValidatorsRoot common.Root
+}
+
+// TestSimulatedCometComponent runs the test suite.
+func TestSimulatedCometComponent(t *testing.T) {
+	suite.Run(t, new(SimulatedSuite))
+}
+
+// SetupTest initializes the test environment.
+func (s *SimulatedSuite) SetupTest() {
+	// Create a cancellable context for the duration of the test.
+	s.Ctx, s.CancelFunc = context.WithCancel(context.Background())
+	s.HomeDir = s.T().TempDir()
+
+	// Initialize the home directory, Comet configuration, and genesis info.
+	cometConfig, genesisValidatorsRoot := simulated.InitializeHomeDir(s.T(), s.HomeDir)
+	s.GenesisValidatorsRoot = genesisValidatorsRoot
+
+	// Start the EL (execution layer) Geth node.
+	elNode := execution.NewGethNode(s.HomeDir, execution.ValidGethImage())
+	elHandle, authRPC := elNode.Start(s.T())
+	s.ElHandle = elHandle
+
+	// Prepare a logger backed by a buffer to capture logs for assertions.
+	s.LogBuffer = new(bytes.Buffer)
+	logger := phuslu.NewLogger(s.LogBuffer, nil)
+
+	// Build the Beacon node with the simulated Comet component.
+	components := simulated.FixedComponents(s.T())
+	components = append(components, simulated.ProvideSimComet)
+	s.TestNode = simulated.NewTestNode(s.T(), simulated.TestNodeInput{
+		TempHomeDir: s.HomeDir,
+		CometConfig: cometConfig,
+		AuthRPC:     authRPC,
+		Logger:      logger,
+		AppOpts:     viper.New(),
+		Components:  components,
+	})
+
+	// Retrieve the simulated Comet service.
+	var cometService *simulated.SimComet
+	err := s.TestNode.FetchService(&cometService)
+	s.Require().NoError(err)
+	s.Require().NotNil(cometService)
+	s.SimComet = cometService
+
+	// Start the Beacon node in a separate goroutine.
+	go func() {
+		_ = s.TestNode.Start(s.Ctx)
+	}()
+
+	// Allow a short period for services to fully initialize.
+	time.Sleep(2 * time.Second)
```

Comment:
q: is there an health end point we can hit here instead of just sleeping?
Does not have to be in this PR, just noting it down.
We don't implement yet the health beacon API, but maybe the cometBFT one can work?